python/whylogs/api/writer/writer.py

```diff
@@ -1,13 +1,32 @@
+import os
 from abc import ABC, abstractmethod
-from typing import Any, Optional, TypeVar
+from typing import Any, Optional, TypeVar, Union
 
 from whylogs.core import DatasetProfileView
 
 T = TypeVar("T", bound="Writer")
 
 
-class Writer(ABC):
+class Writable(ABC):
+    @staticmethod
+    def _safe_open_write(path):
+        """Open `path` for writing, creating any parent directories as needed."""
+        try:
+            os.makedirs(os.path.dirname(path), exist_ok=True)
+        except FileExistsError:
+            pass
+        return open(path, "w")
+
+    @abstractmethod
+    def write(self, **kwargs):
```

Comment:
I would put `def write(self, path: Optional[str]=None)` here to make the `Writer <-> Writable` interface explicit.

Reply:
done!